modifications to taint-and-tolerations page (#34067)
kubernetes/website: content/en/docs/concepts/scheduling-eviction/taint-and-toleration.md

a set of {{< glossary_tooltip text="nodes" term_id="node" >}} (either as a preference or a
hard requirement). _Taints_ are the opposite -- they allow a node to repel a set of pods.

> _Tolerations_ are applied to pods, it does not guarantee that pod will get scheduled

Reviewer: The new text does seem more clear to me than what was already there.  What did you feel was wrong with the existing text?

Reviewer 2: @bradtopol @Huang-Wei please weigh in if this sounds better if we're looking to rephrase the content?

```suggestion
_Tolerations_ are applied to pods. They allow but do not necessarily schedule the pods
```

Author: > @dipankardas011, IMHO the meaning of the sentence has been altered with the change you're proposing. While I agree that brackets add complexity, I don't think that rephrasing it in the way you've suggested is helping with simplifying/clarifying the subject better.

Okay will improve

Reviewer 3: @divya-mohan0209 LGTM, but TBH put "do not necessarily" reads better:

```
Tolerations are applied to pods. They allow (but not necessarily) the pods to be scheduled...
```

a set of {{< glossary_tooltip text="nodes" term_id="node" >}} (either as a preference or a
hard requirement). _Taints_ are the opposite -- they allow a node to repel a set of pods.

> _Tolerations_ are applied to pods, if it matches the Taints in any node it is scheduled, but it does not guarantee as scheduler has other paramters to decide.

Reviewer 2: @dipankardas011 - please may I know what was the concern in adding the suggestion as-is [per the comment](https://github.com/kubernetes/website/pull/34067#discussion_r887682583)?

Author: Because @Huang-Wei was not convinced with the chages that why changed

Reviewer 2: I'm not sure if you're reading the [same comment](https://github.com/kubernetes/website/pull/34067#discussion_r888181117) as I am, however @Huang-Wei please could you clarify which of the suggestions did you sign off on?

> _Tolerations_ are applied to pods, if it matches the Taints in any node it is scheduled, but it does not guarantee as scheduler has other paramters to decide.

Reviewer 4: Building on @divya-mohan0209's proposed language. 

@Huang-Wei Does this answer your concerns about the cause-and-effect relationship of Tolerations and the scheduler?
```suggestion
_Tolerations_ are applied to pods. Tolerations allow the scheduler to schedule pods with matching taints that match the scheduler's other requirements.
```

Reviewer 2: I agree with @zacharysarah and it sounds way better!

Author: Just one doubt aren't we removing that explicit exception?

Reviewer 4: @dipankardas011 Can you say which exception specifically? I'm looking at your original changes and can't immediately see what you mean.

Author: It is not explicitly saying that it may not be scheduled like in the original content where it gives a hint. As it didn't sound correct that's why this PR🙂

Author: if its okay then I can add this suggestion

Reviewer 4: @dipankardas011 Please do add the suggestion.

Author: ```suggestion
_Tolerations_ are applied to pods. Tolerations allow the scheduler to schedule pods with matching taints that meet the scheduler's other requirements. even though it's not guaranteed!
```

Is it okay?

Reviewer 4: @dipankardas011 Thanks, I see what you're going for now. I think one more set of edits can do it. 👍🏻

Reviewer 4: @dipankardas011 Does this work? Adding a link to pod priority and preemption may help others understand the larger context.

If you're okay with this suggestion, go ahead and commit it. We're close!

```suggestion
_Tolerations_ are applied to pods. Tolerations allow the scheduler to schedule pods with matching taints. Tolerations allow scheduling but don't guarantee scheduling: the scheduler also [evaluates other parameters](https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/) as part of its function.
```

Author: That would be a good addition, Thanks👍

Author: So committing this change.

Reviewer 4: @dipankardas011 Yes please.

## Concepts

Basic Syntax
```shell
kubectl taint nodes <node_name> <key>=<value>:<effect>

> ```

Author: Okay
